gadget/gadget_test.go

```diff
@@ -924,7 +934,9 @@ func (s *gadgetYamlTestSuite) TestReadMultiVolumeGadgetYamlValid(c *C) {
 				},
 			},
 		},
-	})
+	}
+	setEnclosingVolumeInStructs(expectedgi.Volumes)
+	c.Assert(ginfo, DeepEquals, expectedgi)
```

Comment:
I'm not sure this can make a difference between a pointer and a copy, at least one test should check that the pointers are the same as the enclosing volume

Reply:
Good point, I have checks for this now